tests/sfbase.py

```diff
@@ -900,3 +918,241 @@ def setUpClass(cls):
 
         if missing_envs:
             raise Exception("set environment variables")
+
+    def get_custom_fields(self, found_catalogs, conn_id):
+        """ List all the custom_fields for each stream"""
+        custom_fields = {}
+        for stream in self.streams_to_test():
+            with self.subTest(stream=stream):
+
+                catalog = [catalog for catalog in found_catalogs if catalog["stream_name"] == stream][0]
+                schema = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])["annotated-schema"]
+                custom_fields[stream] = {key for key in schema['properties'].keys() if key.endswith("__c")}
+        return custom_fields
+
+    def get_non_custom_fields(self, found_catalogs, conn_id):
+        """ List all the non_custom_fields for each stream"""
+        non_custom_fields = {}
+        for stream in self.streams_to_test():
+            with self.subTest(stream=stream):
+                catalog = [catalog for catalog in found_catalogs if catalog["stream_name"] == stream][0]
+                schema = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])["annotated-schema"]
+                non_custom_fields[stream] = {key for key in schema['properties'].keys() if not key.endswith("__c")
+                                             and  schema['properties'][key]['inclusion']!="unsupported"}
```

Comment:
1. There's two spaces between `and` and `schema`
   - There should be one 
2. There's no spaces around `!=` 
   - There should be one on each side

Optional: I personally think if there's an `if` in a comprehension, it should get its own line and every `and` after that too. To me, it helps separate the filtering from the looping
```python
non_custom_fields[stream] = {key 
                             for key in schema['properties'].keys() 
                             if not key.endswith("__c")
                             and schema['properties'][key]['inclusion'] != "unsupported"}
```